agent/config.go

```diff
@@ -151,3 +153,31 @@ func SASLOpt(config *SASLConfig, opts []kgo.Opt) []kgo.Opt {
 	}
 	return opts
 }
+
+// Set the maximum Kafka protocol version to try
+func MaxVersionOpt(version string, opts []kgo.Opt) []kgo.Opt {
+	ver := strings.ToLower(version)
+	ver = strings.ReplaceAll(ver, "v", "")
+	ver = strings.ReplaceAll(ver, ".", "")
+	ver = strings.ReplaceAll(ver, "_", "")
+	verNum, _ := strconv.Atoi(ver)
+	switch verNum {
+	case 330:
+		opts = append(opts, kgo.MaxVersions(kversion.V3_3_0()))
+	case 320:
+		opts = append(opts, kgo.MaxVersions(kversion.V3_2_0()))
+	case 310:
+		opts = append(opts, kgo.MaxVersions(kversion.V3_1_0()))
+	case 300:
+		opts = append(opts, kgo.MaxVersions(kversion.V3_0_0()))
+	case 280:
+		opts = append(opts, kgo.MaxVersions(kversion.V2_8_0()))
+	case 270:
+		opts = append(opts, kgo.MaxVersions(kversion.V2_7_0()))
+	case 260:
+		opts = append(opts, kgo.MaxVersions(kversion.V2_6_0()))
+	default:
+		opts = append(opts, kgo.MaxVersions(kversion.Stable()))
```

Comment:
The trailing 0 can be dropped here -- patch number is always 0 for an API release (with separate patches not introducing new APIs), I wish I knew that before I committed to the kversion package naming.

Reply:
(the only time patch mattered was in the 0.x.x releases -- 0.10.0, 0.10.1, 0.10.2 all introduced new APIs)